drake/systems/plants/RigidBodySystem.h

```diff
@@ -13,24 +69,56 @@ namespace Drake {
     template <typename ScalarType> using OutputVector = Eigen::Matrix<ScalarType,Eigen::Dynamic,1>;
 
     RigidBodySystem(const std::shared_ptr<RigidBodyTree>& rigid_body_tree) : tree(rigid_body_tree) {};
+    RigidBodySystem(const std::string &urdf_filename, const DrakeJoint::FloatingBaseType floating_base_type = DrakeJoint::QUATERNION) {
+      //tree = std::allocate_shared<RigidBodyTree>(Eigen::aligned_allocator<RigidBodyTree>()); // this crashed g++-4.7
+      tree = std::shared_ptr<RigidBodyTree>(new RigidBodyTree());
+      addRobotFromURDF(urdf_filename, floating_base_type);
+    }
     virtual ~RigidBodySystem() {};
 
+    void addRobotFromURDFString(const std::string &xml_string, const std::string &root_dir = ".", const DrakeJoint::FloatingBaseType floating_base_type = DrakeJoint::ROLLPITCHYAW);
+    void addRobotFromURDF(const std::string &urdf_filename, const DrakeJoint::FloatingBaseType floating_base_type = DrakeJoint::QUATERNION);
+
+    // note: will generalize this soon
+    void addForceElement(const std::shared_ptr<RigidBodyPropellor>& prop) { props.push_back(prop); }
+
+    const std::shared_ptr<RigidBodyTree>& getRigidBodyTree(void) { return tree; }
+
     template <typename ScalarType>
     StateVector<ScalarType> dynamics(const ScalarType& t, const StateVector<ScalarType>& x, const InputVector<ScalarType>& u) const {
       using namespace Eigen;
-      const eigen_aligned_unordered_map<const RigidBody *, Matrix<ScalarType, 6, 1> > f_ext;
+      eigen_aligned_unordered_map<const RigidBody *, Matrix<ScalarType, 6, 1> > f_ext;
 
       // todo: make kinematics cache once and re-use it (but have to make one per type)
       auto nq = tree->num_positions;
       auto nv = tree->num_velocities;
+      auto num_actuators = tree->actuators.size();
       auto q = x.topRows(nq);
       auto v = x.bottomRows(nv);
       auto kinsol = tree->doKinematics(q,v);
 
       auto H = tree->massMatrix(kinsol);
       MatrixXd Hinv = H.ldlt().solve(MatrixXd::Identity(nv,nv));
```

Comment:
Why not move the `.solve(` to where the inverse product is actually needed?  Is there a need for storing the actual inverse here?